internal/healthcheck/healthcheck.go

```diff
@@ -0,0 +1,125 @@
+package healthcheck
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-logr/logr"
+	v12 "github.com/openshift/api/config/v1"
+	"github.com/openshift/api/machineconfiguration/v1"
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"time"
+)
+
+// +kubebuilder:rbac:groups=machineconfiguration.openshift.io,resources=machineconfigpools,verbs=list;watch
+// +kubebuilder:rbac:groups=config.openshift.io,resources=clusteroperators,verbs=list;watch
+// +kubebuilder:rbac:groups=operators.coreos.com,resources=clusterserviceversions,verbs=list;watch
+
+var (
+	pollInterval = 3 * time.Second
+	pollTimeout  = 30 * time.Minute
+)
+
+func ClusterServiceVersionReady(c client.Client, l logr.Logger) error {
+	l.Info("Waiting for all ClusterServiceVersion (csv) to be ready")
+
+	err := wait.PollUntilContextTimeout(context.Background(), pollInterval, pollTimeout, true, isClusterServiceVersionReady(c, l))
```

Comment:
With each step having a separate poll, this could take 90 mins to timeout. Consider having a single poll

Reply:
Updated to make it parallel + max 15min of wait for each